lib/Runtime/Types/DictionaryTypeHandler.cpp

```diff
@@ -869,6 +869,81 @@ namespace Js
     }
 
     template <typename T>
+    BOOL DictionaryTypeHandlerBase<T>::DeleteProperty(DynamicObject *instance, JavascriptString *propertyNameString, PropertyOperationFlags propertyOperationFlags)
```

Comment:
@kunalspathak, whether propertyString and propertyId versions call any common code depends on the type of the object and the kind of the type handler. As you might have noticed, there are cases in which the propertyString version falls back to the propertyId after doing a few checks. Those checks are only required in the propertyString version. 
Type handlers that dont support deleting a property by name already call the propertyString version of the base class TypeHandler.

---
In reply to: [78863297](https://github.com/Microsoft/ChakraCore/pull/1542#discussion_r78863297) [](ancestors = 78863297,78818835,78818261,78817010,78791124)